### PR TITLE
BUG: Series(dask.array) GH#38645

### LINKED ISSUE #38645: REGR: passing dask arrays to Series or DataFrame
#### Code Sample, a copy-pastable example

```python
import pandas as pd
import dask.array as da
a = da.ones((12,), chunks=4)
s = pd.Series(a, index=range(12))
print(s.dtype)
```

#### Problem description

This has been detected by `xarray`'s upstream-dev CI ([environment](https://github.com/pydata/xarray/runs/1592176067?check_suite_focus=true#step:5:1177)): with `1.1.3`, the `dtype` is `float64` while on `master` (installed from `scipy-wheels-nightly`) this became `object` (and the series / dataframe contains `dask` scalars). Was that change intentional? Poking around on the merged PR list, this might have been #38563 (not sure, though).

To be clear, for us this only affects test code and since it would compute anyways we can easily work around this by computing the `dask` array before passing it to `pd.Series` or `pd.DataFrame`.

See also pydata/xarray#4717.

cc @TomAugspurger

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

This should close #38645 in conjunction with dask/dask#7888, will hold off on closing the issue until @jsignell confirms.

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
do we want to include in 1.3.1? and include a release note so that users know that constructors now support the numpy array protocol?

**Comment by simonjayhawkins** [comment]
> and include a release note so that users know that constructors now support the numpy array protocol?

maybe also update the docs, although the discussion on what is array-like is ongoing #41807

File: pandas/core/construction.py [hunk 1]
```diff
@@ -560,8 +560,11 @@ def sanitize_array(
             raise TypeError(f"'{type(data).__name__}' type is unordered")
 
         # materialize e.g. generators, convert e.g. tuples, abc.ValueView
-        # TODO: non-standard array-likes we can convert to ndarray more efficiently?
-        data = list(data)
+        if hasattr(data, "__array__"):
```
Comment (jreback) [inline pandas/core/construction.py:563]: can you run this thru L518? where it makes more sense

Comment (jbrockmendel) [reply pandas/core/construction.py:563]: will take some rearrangement (move EA check about ndarray check) but i think doable

**Comment by jreback** [comment]
i suppose this is fine for 1.3.1 and can rearrange in a follow on

if u can add a release note

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Pull request merged**